packages/studio/src/components/NumberPropInput.tsx

```diff
@@ -0,0 +1,41 @@
+import { PropValueType, TypeGuards } from "@yext/studio-plugin";
+import { ChangeEvent, useCallback } from "react";
+import FieldPickerInput from "./FieldPicker/FieldPickerInput";
+
+interface PropInputProps {
+  value: number;
+  onChange: (value: number | string) => void;
+  disabled?: boolean;
+}
+
+/**
+ * Renders the input element of a number prop, that will update the
+ * corresponding prop's value for the active component based on a user's input.
+ */
+export default function NumberPropInput({
+  value,
+  onChange,
+  disabled,
+}: PropInputProps): JSX.Element {
+  const handleChangeEvent = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      const val = Number(e.target.value);
```

Comment:
If val gets parsed to NaN can we prevent onChange from being called?

Reply:
yup exactly!

Reply:
Done! Updated [loom](https://www.loom.com/share/a939e94b04114f9e8f28657a7d4d3ca0?sid=78736e87-28f1-41d2-bf48-cfbc266acaaa) since no more NaN